test/integ/pandas/test_pandas_tools.py

```diff
@@ -255,6 +256,97 @@ def test_write_pandas(
             cnx.execute_string(drop_sql)
 
 
+def test_write_non_range_index_pandas(
+    conn_cnx: Callable[..., Generator[SnowflakeConnection, None, None]],
+    db_parameters: dict[str, str],
+):
+    compression = "gzip"
+    chunk_size = 3
+    quote_identifiers: bool = False
+    auto_create_table: bool = True
+    create_temp_table: bool = False
+    index: bool = False
+
+    # use pandas dataframe with float index
+    n_rows = 17
+    pandas_df = pd.DataFrame(
+        pd.DataFrame(
+            numpy.random.normal(size=(n_rows, 4)),
+            columns=["a", "b", "c", "d"],
+            index=numpy.random.normal(size=n_rows),
+        )
+    )
+
+    # convert to list of tuples to compare to received output
+    pandas_df_data = [tuple(row) for row in list(pandas_df.values)]
+
+    num_of_chunks = math.ceil(len(pandas_df_data) / chunk_size)
+
+    with conn_cnx(
+        user=db_parameters["user"],
```

Comment:
I don't think you need to specify these parameters.

Reply:
I copied the test above, how can I get rid off the params?

Reply:
`with conn_cnx() as cnx:` should be enough